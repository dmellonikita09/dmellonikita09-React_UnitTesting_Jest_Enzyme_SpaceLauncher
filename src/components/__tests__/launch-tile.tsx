import { mount } from 'enzyme';
import React from 'react';
import LaunchTile from '../launch-tile';

describe('Launch Tile', () => {
  it('renders without error', () => {
    const component = mount(
      <LaunchTile
      launch={{
        __typename: 'Launch',
        isBooked: false,
        id: '1',
        mission: { name: 'the first one', __typename: 'Mission', missionPatch: null },
        rocket: { name: 'harambe', __typename: 'Rocket', id: '1' },
      }}
      />
    );
    expect (component).toMatchSnapshot();
  });
});