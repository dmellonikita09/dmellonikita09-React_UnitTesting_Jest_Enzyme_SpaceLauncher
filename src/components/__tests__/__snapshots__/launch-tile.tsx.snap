// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch Tile renders without error 1`] = `
<LaunchTile
  launch={
    Object {
      "__typename": "Launch",
      "id": "1",
      "isBooked": false,
      "mission": Object {
        "__typename": "Mission",
        "missionPatch": null,
        "name": "the first one",
      },
      "rocket": Object {
        "__typename": "Rocket",
        "id": "1",
        "name": "harambe",
      },
    }
  }
>
  <Styled(Link)
    style={
      Object {
        "backgroundImage": "url(iss.jpg)",
      }
    }
    to="/launch/1"
  >
    <Link
      className="css-10q0vj5"
      style={
        Object {
          "backgroundImage": "url(iss.jpg)",
        }
      }
      to="/launch/1"
    >
      <Location>
        <LocationProvider
          history={
            Object {
              "_onTransitionComplete": [Function],
              "listen": [Function],
              "location": Object {
                "hash": "",
                "host": "localhost",
                "hostname": "localhost",
                "href": "http://localhost/",
                "key": "initial",
                "origin": "http://localhost",
                "pathname": "/",
                "port": "",
                "protocol": "http:",
                "search": "",
                "state": null,
              },
              "navigate": [Function],
              "transitioning": false,
            }
          }
        >
          <a
            className="css-10q0vj5"
            href="/launch/1"
            onClick={[Function]}
            style={
              Object {
                "backgroundImage": "url(iss.jpg)",
              }
            }
          >
            <h3>
              the first one
            </h3>
            <h5>
              harambe
            </h5>
          </a>
        </LocationProvider>
      </Location>
    </Link>
  </Styled(Link)>
</LaunchTile>
`;
